### ISSUE TITLE
python3 include server fails with UnicodeDecodeError

### ISSUE THREAD

**Issue by gjkloosterman** [distcc/distcc#224]
When using Python3, the include server fails with an internal error on encountering non-UTF-8 data:
```
WARNING include server: Preprocessing locally. Include server internal error: '<class 'UnicodeDecodeError'>: ('utf-8', b'......
```
In my case, an included header happened to contain a comment with an iso-8859-1 encoded copyright sign.

The problem disappeared after I reverted to Python2 by applying #182

**Comment by martinm82** [comment]
In case you still want to make use of Python3 I applied in my fork a possible fix for that error. I could not easily switch to Python 2 so decided to fix it.

https://github.com/tt-gf/distcc/commit/87c46e379885acb4ece84e3e06bc20ccae42a24c

I know the solution is not perfect but I did not want to introduce the Python 'magic' module as otherwise the user would need to install more dependencies.

**Comment by afbjorklund** [comment]
@martinm82 : can you make a pull request ? I guess _someone_ still wants `python3` on master.

**Comment by martinm82** [comment]
@afbjorklund, found more unicode issues. So once I verify it with a full build I will create a PR

**Comment by shawnl** [comment]
Are not all the characters we are interested in parsing ASCII, so can't we just ignore non-ASCII bytes, and pass them through as-is? Does windows-1252 or ISO-8859-1 give us that? (UTF-8 does not because of invalid sequences) that way we add no dependencies.

**Comment by shawnl** [comment]
should be fixed with 23d36291
